pyani_plus/private_cli.py

```diff
@@ -558,10 +558,16 @@ def log_dnadiff(
     # Should we add --maxmatch (nucmer) and -m (deltafilter) parameters?
     # These are default parameters used in workflows
 ) -> int:
-    """Log a single pyANI-plus ANIm pairwise comparison (with nucmer) to the database."""
+    """Log a single pyANI-plus dnadiff pairwise comparison (with nucmer) to the database."""
     # Assuming this will match as expect this script to be called right
     # after the computation has finished (on the same machine)
-    dnadiff_tool = tools.get_dnadiff()
+    # We don't actually call the tool dnadiff (which has its own version),
+    # rather we call nucmer, delta-filter, show-diff and show-coords from mumer
+    tool = tools.ExternalToolData(
+        exe_path=tools.get_delta_filter().exe_path,
+        # As a proxy for the missing delta-filter version
+        version=tools.get_nucmer().version,
+    )
```

Comment:
I started work on adding dnadiff to the public API, and noticed an inconsistency here.

It seems meaningless and misleading to record the actual dnadiff version, because we don't actually use it:

```
$ dnadiff --version
dnadiff 
DNAdiff version 1.3
```

Rather we call nucmer, delta-filter, show-diff and show-coords from MUMmer - and as I understand it re-implement the logic of dnadiff to get the ANI etc.

The other MUMmer tools effectively all share the single version reported by nucmer:

```
$ nucmer --version
nucmer 
NUCmer (NUCleotide MUMmer) version 3.1
```

So, what should we record as the "program" and "version" fields? One of the MUMer tools (say "delta-filter") and the version via nucmer ("3.1" here)?

Reply:
I'm happy to change this to `nucmer` as other tool slike `show-coords` and `show-diff` don't have a version anyways.

Reply:
That would be simpler - and the pipeline starts with nucmer anyway 👍